### PR TITLE
Special installation aspects (#358)

### LINKED ISSUE #358: Functions to configure commonly-used settings - Set Static IP, Configure Proxy, Open-VM-Tools
- It would be very convenient to have the ability to set a static ip address as part of the installation. Installs are often started without a connection to a dhcp network.
- It would be very useful to have the ability to set and enable a proxy for internet access when on a private ip network.  Otherwise, installation of updates, Rock-Ons, etc is not possible. I envision this as a button on the Services page.
- VMware is a common hosting platform to install Rockstor and the Open-VM-Tools package can enable increased management function.  It would be very convenient to simplify installation of this package, either as part of the installation or from the GUI after installation.

### PR DESCRIPTION
**Pull request by Hooverdan96**
Fixes #358 

### This pull request's proposal
Add a section for commonly-used "special" aspects of Rockstor installations.

### Checklist
- [X] With the proposed changes no Sphinx errors or warnings are generated.
- [X] I have added my name to the AUTHORS file, if required (descending alphabetical order).

### CODE REVIEW

**Comment by Hooverdan96** [comment]
FYI, the `linkcheck_ignore` was updated for the three failures in another PR (#500) that's currently open.

**Review by phillxnet — CHANGES_REQUESTED** [review]
@Hooverdan96 This is a nice addition, with lots of details. Thanks.

But for me this is entirely miss-placed in our generic installer-howto. It pertains to likely less than 1% of users, and looks super scary to the 99% who will look at this doc, hopefully scrolling through to see what they may encounter. Viewing this last section would put me off. We are, up to this point. Super simple. I'm strongly against this addition in this how-to. But the contents is great for folks who are CLI happy with advanced requirements. Not 99% of installer users as I see it. Otherwise they would not be using a Web-UI NAS setup but hand-crafting their own non-limited configurations on a linux base of their choice.

How about this same contents is put in it's own how-to: and we have a note at the end of the installer doc that in-turn links to this new Special Installation Scenarios, with the note highlighting that the vast majority of users will not need to reference the 'Special'.

We absolutely must keep our existing how-to as short as possible. And entirely void of this type of off-putting detail; was my thinking. I.e. akin to @FroggyFlox moving my prior downloads page instructions on installing the rpms on vanilla openSUSE:

https://rockstor.com/docs/howtos/rpm_install.html

I crammed in details on the download page, where-as they deserved their own how-to.
I think similarly about this highly advanced extension to our installer how-to.

So in-brief I don't think this belongs in the generic installer how-to as it does not pertain to our installer: but to advanced setup pertaining to a an extreme minority of users. Ergo we have a single paragraph note at most referencing this content but in its own how-to. Folks can then reference that doc if need-be. Keeping the existing how-to focused to our installer and how to use it. Not all possible variations of what configuration may pertain to niche and in some cases with this additional info unsupported installs (VM/hypervisor host).

Lets see what it looks like in its own how-to: with a single paragraph note leading to it from the installer. Where that note indicates why you may need to click on the 'Special ...'.  But little else. We absolutely must not grow this base installer how-to more than is absolutely necessary was my thinking here.

However: the write-up of what you've addressed is, as is normal for you, excellent  :) .

What do your think? Are you OK moving this to its own how-to? We could have that how-to listed right next to this installer: if that helps to group stuff together. And as it also pertains to actually installing, possibly next to our existing:

Install on Vanilla openSUSE/SuSE SLES

I.e. while there we could have our installer-howto also added to the how-to entries: resulting in for example:

- Rockstor’s “Built on openSUSE” installer (not moved, just added as a how-to link)
- Special Installation Scenarios (your advanced/unsupported options/notes presented here)
- Install on Vanilla openSUSE/SuSE SLES (@FroggyFlox advancement/abstraction on my original downloads page bloat).

That way we are helping to group together progressively more advanced install doc entries. Where the first servers the vast majority, but all can easily lead into the more advanced option as needed.

**Comment by Hooverdan96** [comment]
@phillxnet, @FroggyFlox: take a look at this. After rebasing, I broke out the special install scenarios. I also took your "hint" in #506 to adjust the text to semantic linefeeds within the documents I touched.

File: installation/special-install-scenarios-howto.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+VMWare guest tools installation
+-------------------------------
+
+As VMWare is a commmon virtualization package that is used by the Rockstor (alongside with others VirtualBox, Proxmox, etc.), there is a useful toolset available that helps with various aspects of the integration between
+the host and guest among other tools (similar guest extensions exist under VirtualBox as well): **open-vm-tools**.
+
+There is the option to build this package into the custom installer, so it becomes part of the initial installation routine.
+When building a customer installer to include it, using the `Rockstor installer <https://github.com/rockstor/rockstor-installer>`_
+the package is already contained in the **rockstor.kiwi** file, albeit commented out.
```
Comment (phillxnet) [inline installation/special-install-scenarios-howto.rst:85]:
```suggestion
This or any package can be included into a custom installer, making them available during the initial installation.
For building a custom installer see our `rockstor-installer <https://github.com/rockstor/rockstor-installer>`_ repo.
The `open-vm-tools` package is already contained in the **rockstor.kiwi** file, albeit commented out.
```

File: installation/special-install-scenarios-howto.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+.. code-block:: console
+
+   nmcli connection up "enp0s3"
+
+or reboot the system.
+
+.. note::
+   The above is meant for setting up a specific connection **before** the installation/configuration is completed using Rockstor's WebUI.
+   Once the Rockstor installation is complete, a static IP can be assigned from the WebUI. Under most circumstances neither should be necessary, as most routers today offer IP address reservation that will effectively result in a static IP address assigned to the Rockstor appliance (which by default uses `dhcp` to automatically obtain an IP address from the gateway).
```
[the remark this answers is not in the document]
Comment (Hooverdan96) [reply installation/special-install-scenarios-howto.rst]: I've rearranged the section a bit based on your recommendation and did some rewording.

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for applying the requested changes. Can I now look to this again, i.e. are you done with the changes?

Thanks for this monster recent docs effort by the way. Much appreciated. And my apologies to taking so long to track all your changes :) .

**Comment by Hooverdan96** [comment]
you're very welcome, and no worries on it taking more time, you do have a life :).

Yes, you can review the changes now.

**Review by phillxnet — APPROVED** [review]

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for what is a definite improvement in installer doc coverage. And for accommodating my suggestions re rearrangements.

All looks dandy now so I'll merge a get to publishing shortly.

**Comment by phillxnet** [comment]
PR product PRODUCTION published.

**Pull request merged**